src/lib/FeedbackForm.svelte

```diff
@@ -1,53 +1,135 @@
 <script lang="ts">
 
+  import FormMessage from './FormMessage.svelte';
+  import { slide } from 'svelte/transition';
   import '@shoelace-style/shoelace/dist/components/button/button.js';
   import '@shoelace-style/shoelace/dist/components/input/input.js';
   import '@shoelace-style/shoelace/dist/components/textarea/textarea.js';
   import { serialize } from '@shoelace-style/shoelace/dist/utilities/form.js'
+  import '@shoelace-style/shoelace/dist/components/alert/alert.js';
    
   let userURL: string = location.href;
   let userAgent: string = navigator.userAgent;
+
+  //takes long string output of document.cookie and splits it into a usable javascript object
+  let cookies: object = document.cookie
+  .split(';')
+  .map(cookie => cookie.split('='))
+  .reduce((accumulator, [key, value]) => ({ ...accumulator, [key.trim()]: decodeURIComponent(value) }), {});
+
+  //if user isn't logged in, HTstatus cookie won't exist
+  let userAuthStatus: string = cookies.HTstatus || 'not logged in';
+
+  let postResponseStatusCode;
+
+  // when true, spinner on submit button animates
+  let loading = false;
+  // when true, hides the element (in this case, the form)
+  let hidden = false;
+  // when true, shows the success/failure alert message
+  let submitted = false;
   
   const onSubmit = async () => {
+    loading = true;
     //shoelace serializer for turning FormData into JSON
     const form = document.querySelector('form')
     const data = serialize(form)
 
-    const res = await fetch('http://localhost:5000/api', {
-          method: 'POST', 
-            body: JSON.stringify(data),
-            headers: {
-                'Content-Type': 'application/json',
-                'Accept': 'application/json'
-            }
-        })
-
-        const json = await res.json()
-        console.log(JSON.stringify(json))
+    try {
+      const res = await fetch('http://localhost:5000/api', {
+            method: 'POST', 
+              body: JSON.stringify(data),
+              headers: {
+                  'Content-Type': 'application/json',
+                  'Accept': 'application/json'
+              }
+          })
+
+      const json = await res.json()
+      postResponseStatusCode = res.status;
+      loading = false;
+      submitted = true;
+
+      if (res.ok) {
+        hidden = true;
+        console.log(`request created in service desk ${json.serviceDeskId}: ${json.issueKey}`)
+        console.log('status code', postResponseStatusCode)
+      } else {
+        throw new Error(`There was an error posting the form: ${res.status}, ${res.statusText}`)
+      }
+    } catch(error) {
```

Comment:
You're capturing the `fetch` response error? You can do 

```
fetch()
.then()
.catch((error) => {})
```

Reply:
The annoying thing about `fetch` is that `error` doesn't fire for HTTP status errors: that's why you still need to do `response.ok`. `error` _should_ fire when you can't reach the server (as I understand; promises are awesome until they're not)

Reply:
Might be best if I refactor this one function into at least two: one that does the fetch (and handles those errors) and one that does the onSubmit stuff. If nothing else, it will be cleaner to read/understand!